### PR TITLE
Refactors exports and pirate loot gathering.

### PR DESCRIPTION
**Pull request by AnturK**
Pirates will now use cargo hold terminal and pad to dump their ill-gotten gains. Including hostages.

Refactors exports to be less dependant on cargo state.

Pirate shuttle remapped a bit by @WJohn.

### CODE REVIEW

**Comment by WJohn** [comment]
The pirate shuttle changes in question.

**Comment by 81Denton** [comment]
What happens to hostages when they're exported, do they get QDEL'd?
Also thank you for making the vault larger!

**Comment by AnturK** [comment]
Yep, hostages just die when sold. I was considering storing them in ninja prison instead but any way they're not able to be rescued at that point.

**Comment by Kagus** [comment]
You might theoretically be able to sell "dead" hostages by reviving killed crewmembers via surgery/strange reagent, but considering the hoops a pirate would need to jump through in order to do that, I don't think that's a particularly important loophole.

While it is a bit sad to not have the same kind of booty vault that you can prance around in, this method does seem a lot more resistant to crate clogging, crew griefing, and space assistants coming to re-steal your honorably pillaged loot.

**Comment by nicbn** [comment]
Pirate shuttle doesn't need atmos, pirates are skeletons. Just give them a O2 cannister so they can replace air if they want to bring in hostages.

**Comment by WJohn** [comment]
I believe antur said they wanted to remove the whole skeleton thing because it invalidated most of everything on their ship. The only reason why they were skeletons at all was because people didn't realize the clothes they spawned in were space proof -- and the reason for that is because a fucking pirate hat and open leather coat does not really intuitively point that they're space proof at all.

**Comment by nicbn** [comment]
Actually it was me who made them skeletons and it wasn't because of skeletons being space proof, it was because of aesthetics - you know, rattle me bones and all.

**Comment by Kagus** [comment]
I figure it fits; they're trying to recollect treasure of equal value to what they looted from the Space-Aztec ruin that ended up cursing them when they spent it all on booze and liver replacement surgery. In the meantime, they're cursed with extra spookiness and the constant derision of every proper antagonist. Also mutiny. So much mutiny...

**Comment by fizzle7** [comment]
Do the external airlocks have an air vent? It needs one otherwise space just pulls everything outside

**Comment by ninjanomnom** [comment]
It uses a cycle airlock

File: code/modules/cargo/exports.dm [hunk 1]
```diff
@@ -122,26 +119,39 @@ Credit dupes that require a lot of manual work shouldn't be removed, unless they
...
+/datum/export/proc/sell_object(obj/O, datum/export_report/report, dry_run = TRUE, allowed_categories = EXPORT_CARGO , apply_elastic = TRUE)
+	var/the_cost = get_cost(O, allowed_categories , apply_elastic)
 	var/amount = get_amount(O)
-	total_cost += the_cost
+
+	if(amount <=0 || the_cost <=0)
+		return FALSE
+	
+	report.total_value[src] += the_cost
+	//total_cost += the_cost
```
Comment (Cyberboss) [inline code/modules/cargo/exports.dm]: ?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/cargo/exports.dm (hunk 1 above)
Comment (AnturK) [reply code/modules/cargo/exports.dm]: No idea why i left this commented. Will remove in a moment.

**Comment by AnturK** [comment]
Please take a glance at map contents when reviewing this was blind conflict resolution.

**Pull request merged**